### PR TITLE
Add toggle for physical hardware shift+space layout switch shortcut

### PR DESCRIPTION
**Pull request by antonok-edm**
Issue #111 mentioned that the current shift+space shortcut for switching layouts on physical keyboards was unintuitive, and it is inconvenient when typing capital letters, parentheses, underscores, etc.

This PR adds a toggle in the hardware keyboard settings to disable that shortcut.

### CODE REVIEW

**Comment by shaggy16** [comment]
Add

**Comment by menny** [comment]
Thanks.
Can you add another settings to control ALT+SHIFT too?

**Comment by antonok-edm** [comment]
@menny I can certainly do that, but I'd like to have a little more background information before I make any changes there. I opened #1467 for further discussion.

I also propose completing that in a separate PR, since the functionality seems to be independent of the Shift+Space shortcut.

File: app/src/main/res/values-en/strings.xml [hunk 1]
```diff
@@ -174,6 +174,12 @@
     <string name="hide_soft_when_physical_off_summary">Show soft keyboard when physical keyboard is
         used
     </string>
+    <string name="enable_alt_space_language_shortcut">Alt + Space switches languages</string>
```
Comment (menny) [inline app/src/main/res/values-en/strings.xml]: You don't need to specify the strings for `values-en/strings.xml`, crowd-in service generates those automatically (directly copies from `values/strings.xml`).

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -174,6 +174,12 @@
     <string name="hide_soft_when_physical_off_summary">Show soft keyboard when physical keyboard is
         used
     </string>
+    <string name="enable_alt_space_language_shortcut">Alt + Space switches languages</string>
```
Comment (menny) [inline app/src/main/res/values/strings.xml]: I believe we should use uppercase _SPACE_, and not _Space_. What do you think?

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -174,6 +174,12 @@
...
+    <string name="enable_alt_space_language_shortcut_on_summary">Alt + space switches to the next available layout.</string>
```
Comment (menny) [inline app/src/main/res/values/strings.xml]: Maybe indicate that this is affecting only physical keyboards?
`Hardware ALT + SPACE switches to the next available layout.`

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (antonok-edm) [reply app/src/main/res/values/strings.xml]: I was trying to be consistent with the style of some of the other config options -- e.g. "Space switches keyboards": "Switch from symbols/numbers to alphabet when pressing space".
I've found a mix of "Space" and "space", but I don't see any "SPACE".

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (antonok-edm) [reply app/src/main/res/values/strings.xml]: It is in the "Physical Keyboard Tweaks" section, but the headers are fairly small so I think that makes sense.

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (menny) [reply app/src/main/res/values/strings.xml]: Then capitalizrd `Space`. It's a name of  key.

**Comment by antonok-edm** [comment]
@menny Should be all set.

**Comment by menny** [comment]
Thanks!

**Comment by poet-man** [comment]
Not being familiar with it … how/where can I switch off that shift-space for changing layouts?
I happen to activate it accidentally very often.
Thanks

**Comment by antonok-edm** [comment]
@poet-man The option is called 'Shift + Space switches languages'; you can find it under Language > Tweaks and More > Hardware Keys Mapping > Tweaks > Physical Keyboard Tweaks

**Comment by poet-man** [comment]
Thanks a lot

Am 6. November 2018 22:03:31 schrieb Anton Lazarev <notifications@github.com>:

**Pull request merged**